EIPS/eip-5548.md

```diff
@@ -0,0 +1,126 @@
+
+However, as intermediary platforms willfully circumvent social codes established by NFT creators, it is within the spirit of crypto to provide technological tools to deny allowances to those intermediaries.
+
+It should be noted that this extension will not (nor does it aim to) stamp out any nefarious behaviour (such as royalty-dodging) from token holders. The position of this EIP is that any restriction to owner's ability to `Transfer` their NFTs would be overreaching. And of course, denyListed platforms could deploy new smart contracts, placing creators in the position of playing "whack-a-mole" to add new deployments to the denyList. 
+
+Nonetheless, there is benefit to creating some punitive friction for bad actors, and to implementing an EIP-721 extension which signals creator's intention to exert control over the monetization of their creations.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+Our denyList is a single new mapping of operator addresses, paired with true/false values. An event is emitted when an address is added to the denyList.
+
+```solidity
+
+mapping (address => bool) public denyList;
+event OperatorDenied(address badOperator);
+
+```
+
+Addresses may be added to the denyList.
+
+
+
+```solidity
+
+function updateDenyList(address badOperator, bool status) public onlyOwner {
+       
+        require(
+            _msgSender() == owner ,
+            "ERC721: approve caller is not token owner nor approved for all"
+        );
+
+        _addToDenyList(badOperator, status);
+    }
+
+function _updateDenyList(address badOperator, bool status) internal virtual {
+        denyList[badOperator] = status;
+        emit OperatorDenied(badOperator);
+    }
+
+```
+
+
+The following functions in the standard OpenZeppelin `ERC721.sol` are slightly modified.
+
+```solidity
+
+function setApprovalForAll(address operator, bool approved) public virtual override {
+        require(denyList[operator] == false, "Operator has been denied by contract owner.");   //ADD THIS LINE ONLY
+        _setApprovalForAll(_msgSender(), operator, approved);
+    }
+    
+function approve(address to, uint256 tokenId) public virtual override {
+      address owner = ERC721.ownerOf(tokenId);
+      require(to != owner, "ERC721: approval to current owner");
+
+      require(
+          _msgSender() == owner || isApprovedForAll(owner, _msgSender()),
+          "ERC721: approve caller is not token owner nor approved for all"
+      );
+      require(denyList[to] == false, "Operator has been denied by contract owner."; //ADD THIS LINE ONLY
+      _approve(to, tokenId);
+  }
+    
+ ```
```

Comment:
Do we need a getter to ask the contract whether a particular marketplace has been blocked.